test/extensions/filters/common/ext_authz/check_request_utils_test.cc

```diff
@@ -127,45 +154,50 @@ TEST_F(CheckRequestUtilsTest, BasicHttpWithFullBody) {
                          Http::Headers::get().EnvoyAuthPartialBody.get()));
 }
 
-// Verify that createHttpCheck extract the proper attributes from the http request into CheckRequest
-// proto object.
-TEST_F(CheckRequestUtilsTest, CheckAttrContextPeer) {
-  Http::TestHeaderMapImpl request_headers{{"x-envoy-downstream-service-cluster", "foo"},
-                                          {":path", "/bar"}};
-  envoy::service::auth::v2::CheckRequest request;
-  EXPECT_CALL(callbacks_, connection()).WillRepeatedly(Return(&connection_));
-  EXPECT_CALL(connection_, remoteAddress()).WillRepeatedly(ReturnRef(addr_));
-  EXPECT_CALL(connection_, localAddress()).WillRepeatedly(ReturnRef(addr_));
-  EXPECT_CALL(Const(connection_), ssl()).WillRepeatedly(Return(&ssl_));
-  EXPECT_CALL(callbacks_, streamId()).WillRepeatedly(Return(0));
-  EXPECT_CALL(callbacks_, streamInfo()).WillRepeatedly(ReturnRef(req_info_));
-  EXPECT_CALL(callbacks_, decodingBuffer()).Times(1);
-  EXPECT_CALL(req_info_, protocol()).WillRepeatedly(ReturnPointee(&protocol_));
+// Verify that createHttpCheck extract the attributes from the http request into CheckRequest
+// proto object and Uri SAN is used as principal if present.
+TEST_F(CheckRequestUtilsTest, CheckAttrContextPeerUriSans) {
+  expectBasicHttp();
+
   EXPECT_CALL(ssl_, uriSanPeerCertificate()).WillOnce(Return(std::vector<std::string>{"source"}));
   EXPECT_CALL(ssl_, uriSanLocalCertificate())
       .WillOnce(Return(std::vector<std::string>{"destination"}));
 
+  callHttpCheckAndValidateRequestAttributes();
+}
+
+// Verify that createHttpCheck extract the attributes from the http request into CheckRequest
+// proto object and Dns SAN is used as principal if Uri SAN is absent.
```

Comment:
Nit: s/Dns/DNS, s/http request/HTTP rquest, etc. here and elsewhere in comments. Same with Uri -> URI in comments.